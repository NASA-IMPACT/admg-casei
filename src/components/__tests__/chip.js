import React from "react"
import renderer from "react-test-renderer"
import Chip from "../chip"

const testActionId = "4c5e9dc3-38ed-40cb-b6ea-cacf96057b53"
const testId = "filter"
const testLabel = "Weather"

describe("Chip", () => {
  it("renders without action", () => {
    const tree = renderer
      .create(<Chip id={testId} label={testLabel} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders with action", () => {
    const tree = renderer
      .create(
        <Chip
          id={testId}
          label={testLabel}
          actionId={testActionId}
          removeAction={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
