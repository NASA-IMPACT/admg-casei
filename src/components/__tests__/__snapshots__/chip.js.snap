// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders with action 1`] = `
.c0.c0:hover {
  opacity: 0.64;
}

.c0.c0:active {
  -webkit-transform: translate(0,1px);
  -ms-transform: translate(0,1px);
  transform: translate(0,1px);
}

<div
  data-cy="filter-chip"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#303641",
      "borderRadius": "0.2rem",
      "color": "hsla(0,0%,100%,0.9)",
      "display": "flex",
      "margin": "0.25rem 0.5rem",
      "padding": "0 0.5rem",
    }
  }
>
  <small
    title=""
  >
    Weather
  </small>
  <button
    className="c0"
    data-cy="remove-filter"
    onClick={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "hsla(0,0%,100%,0.9)",
        "cursor": "pointer",
        "flexGrow": 0,
        "marginLeft": "0.5rem",
        "verticalAlign": "middle",
      }
    }
    type="button"
  >
    <span
      aria-label="remove-filter-icon"
      role="img"
    >
      <svg
        height="12px"
        viewBox="0 0 16 16"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="hsla(0,0%,100%,0.9)"
          points="13.707,3.707 12.293,2.293 8,6.586 3.707,2.293 2.293,3.707 6.586,8 2.293,12.293 3.707,13.707 8,9.414  12.293,13.707 13.707,12.293 9.414,8"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Chip renders without action 1`] = `
<div
  data-cy="filter-chip"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#303641",
      "borderRadius": "0.2rem",
      "color": "hsla(0,0%,100%,0.9)",
      "display": "flex",
      "margin": "0.25rem 0.5rem",
      "padding": "0 0.5rem",
    }
  }
>
  <small
    title=""
  >
    Weather
  </small>
</div>
`;
